libexec/rtld-elf/rtld.c

```diff
@@ -1883,7 +1883,12 @@ digest_phdr(const Elf_Phdr *phdr, int phnum, dlfunc_t entry, const char *path)
     }
 
 #ifdef __CHERI_PURE_CAPABILITY__
-   obj->relocbase = cheri_setbounds(obj->relocbase, obj->mapsize);
+    if (cheri_bytes_remaining(obj->relocbase) > CHERI_REPRESENTABLE_LENGTH(
+	obj->mapsize)) {
```

Comment:
rtld.c uses really messed up indentation, but I think this is consistent with the rest of the file...